user_manager/api/v1/views.py

```diff
@@ -0,0 +1,419 @@
+
+            * results: a list of users under a manager:
+
+                * id: The user id for a user.
+
+                * email: Email address of user.
+
+        **Example GET Response**
+
+            GET /api/user_manager/v1/reports/edx@example.com/
+
+            {
+                "count": 99,
+                "next": "https://courses.org/api/user_manager/v1/reports/edx@example.com/?page=2",
+                "previous": null,
+                "results": {
+                    {
+                        "email": "staff@example.com",
+                        "id": 9
+                    },
+                    { ... }
+                }
+            }
+
+        **Example POST Response**
+
+            POST /api/user_manager/v1/reports/edx@example.com/ {
+                "email": "user@email.com"
+            }
+
+            {
+                "email": "user@email.com"
+                "id": 11
+            }
+
+        **Example DELETE Response**
+
+            DELETE /api/user_manager/v1/reports/edx@exmaple.com/
+
+            DELETE /api/user_manager/v1/reports/edx@exmaple.com/?report=some@user.com
+
+    """
+
+    serializer_class = ManagerReportsSerializer
+
+    def get_queryset(self):
+        """
+        Return queryset with username filter.
+        """
+
+        username = self.kwargs['username']
+        return _filter_by_manager_id(UserManagerRole.objects, username)
+
+    def perform_create(self, serializer):
+        """
+        Query all object needed by serializer.
+        """
+
+        manager_id = self.kwargs['username']
+        email = serializer.validated_data.get('user', {}).get('email')
+
+        try:
+            user = User.objects.get(email=email)
+        except User.DoesNotExist:
+            raise NotFound(detail='No user with that email')
+
+        if '@' in manager_id:
+            try:
+                manager_user = User.objects.get(email=manager_id)
+            except User.DoesNotExist:
+                serializer.save(user=user, unregistered_manager_email=manager_id)
+                return
+        else:
+            manager_user = User.objects.get(username=manager_id)
+
+        serializer.save(manager_user=manager_user, user=user)
+
+    def delete(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Delete one or all reports for provided manager.
+        """
+
+        user = request.query_params.get('user')
+        queryset = _filter_by_user_id(self.get_queryset(), user)
+        queryset.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class UserManagerListView(ManagerViewMixin, ListCreateAPIView):
+    """
+        **Use Case**
+
+            * Get a list of the managers that a user directly reports to.
+
+            * Add a manger for a user.
+
+            * Remove all managers for a user, or remove a single manager for a user.
+
+        **Example Request**
+
+            GET /api/user_manager/v1/managers/{user_id}/
```

Comment:
I think that `v1/users/:username/managers/` makes a lot more sense as a RESTful URL pattern for this API. Generally the ID part should be of the type of the preceding noun, so with `/api/foo/:id`, the ID should be a foo ID.